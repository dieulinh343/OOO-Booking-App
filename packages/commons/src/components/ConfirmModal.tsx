import React, { useState } from 'react';
import Modal, { FooterType } from './Modal';

export function ConfirmModal({
  waitConfirm,
  onConfirm,
  onCancel,
  onModalClose,
  headerText,
  bodyText,
  primaryButtonText,
  isDangerous = false,
  centered = false,
  ...rest
}: {
  waitConfirm?: boolean;
  onConfirm: () => void;
  onCancel: () => void;
  onModalClose: () => void;
  headerText: string;
  bodyText: string;
  primaryButtonText: string;
  isDangerous: boolean;
  centered: boolean;
  rest?: object;
}) {
  const [submitting, setSubmitting] = useState(false);

  return (
    <Modal
      id="modal-confirm"
      centered={centered}
      closable={false}
      size="small"
      headerText={headerText || 'Confirmation'}
      body={(
        <p>{bodyText}</p>
      )}
      primaryButtonText={submitting ? 'SUBMITTING...' : (primaryButtonText || 'YES')}
      onClickPrimaryButton={async () => {
        if (waitConfirm) {
          setSubmitting(true);
          await onConfirm();
        } else {
          onConfirm();
        }

        onModalClose();
      }}
      secondaryButtonText="NO"
      onClickSecondaryButton={() => {
        onCancel && onCancel();
        onModalClose();
      }}
      primaryButtonVariant={isDangerous ? 'negative' : 'primary'}
      footerType={FooterType.DOUBLE}
      disablePrimaryButton={submitting}
      disableSecondaryButton={submitting}
      {...rest}
    />
  );
}
export default ConfirmModal;
