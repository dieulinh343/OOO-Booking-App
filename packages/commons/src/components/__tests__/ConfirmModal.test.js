import React from 'react';
import { shallow } from 'enzyme';
import { ConfirmModal } from '../ConfirmModal';

describe('components/ConfirmModal', () => {
  let wrapper;
  let props;

  const setup = () => {
    wrapper = shallow(<ConfirmModal {...props} />);
    wrapper.update();
  };

  beforeEach(() => {
    props = {
      onConfirm: jest.fn(),
      onModalClose: jest.fn(),
      onCancel: jest.fn(),
      bodyText: 'sample body',
    };
  });

  it('should render ConfirmModal with isDangerous being passed as true', () => {
    props = {
      ...props,
      isDangerous: true,
    };
    setup();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.props().primaryButtonVariant).toEqual('negative');
  });

  it('should render successfully', () => {
    setup();
    expect(wrapper.length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.props().primaryButtonVariant).toEqual('primary');
  });

  it('should call onConfirm when clicking primary button', () => {
    setup();
    wrapper.props().onClickPrimaryButton();
    expect(props.onConfirm).toHaveBeenCalledTimes(1);
  });

  it('should call onCancel and onModalClose when clicking secondary button', () => {
    setup();
    wrapper.props().onClickSecondaryButton();
    expect(props.onCancel).toHaveBeenCalledTimes(1);
    expect(props.onModalClose).toHaveBeenCalledTimes(1);
  });
});
