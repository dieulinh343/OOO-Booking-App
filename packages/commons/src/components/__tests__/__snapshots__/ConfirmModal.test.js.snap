// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ConfirmModal should render ConfirmModal with isDangerous being passed as true 1`] = `
<BaseModal
  body={
    <p>
      sample body
    </p>
  }
  centered={false}
  closable={false}
  disablePrimaryButton={false}
  disableSecondaryButton={false}
  footerType="double"
  headerText="Confirmation"
  id="modal-confirm"
  onClickPrimaryButton={[Function]}
  onClickSecondaryButton={[Function]}
  primaryButtonText="YES"
  primaryButtonVariant="negative"
  secondaryButtonText="NO"
  secondaryButtonVariant="secondary"
  showFooter={true}
  showHeader={true}
  size="small"
/>
`;

exports[`components/ConfirmModal should render successfully 1`] = `
<BaseModal
  body={
    <p>
      sample body
    </p>
  }
  centered={false}
  closable={false}
  disablePrimaryButton={false}
  disableSecondaryButton={false}
  footerType="double"
  headerText="Confirmation"
  id="modal-confirm"
  onClickPrimaryButton={[Function]}
  onClickSecondaryButton={[Function]}
  primaryButtonText="YES"
  primaryButtonVariant="primary"
  secondaryButtonText="NO"
  secondaryButtonVariant="secondary"
  showFooter={true}
  showHeader={true}
  size="small"
/>
`;
